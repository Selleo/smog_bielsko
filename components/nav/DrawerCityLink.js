import React, { Component } from 'react';
import { StyleSheet, TouchableHighlight, Text } from 'react-native';

export default class DrawerCityLink extends Component {
  static get defaultProps() {
  }

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <TouchableHighlight
        onPress={() => this.props.closeDrawer(this.props.station.id) }
        style={this.isActive() ? styles.active : null}
        underlayColor='#aca'
      >
        <Text style={styles.text}>
          {this.props.station.name}
        </Text>
      </TouchableHighlight>
    )
  }

  isActive() {
    return this.props.station.id == this.props.getCurrentStationId();
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 16,
    color: '#444',
    paddingHorizontal: 20,
    paddingVertical: 10
  },
  active: {
    backgroundColor: '#aca'
  }
})
