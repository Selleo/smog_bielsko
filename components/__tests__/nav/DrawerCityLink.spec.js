import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import DrawerCityLink                 from '../../nav/DrawerCityLink';

const originalExpect = global.expect;

chai.use(dirtyChai);

let wrapper, props, closeDrawerSpy;

describe('DrawerCityLink', () => {
  beforeEach(() => {
    props = {
      getCurrentStationId: () => 1,
      station: { id: 1, name: 'Zywiec' },
      closeDrawer: (stationId) => closeDrawerSpy = stationId
    };

    wrapper = shallow(<DrawerCityLink {...props}/>);
  });

  it('renders correctly', () => {
    originalExpect(wrapper).toMatchSnapshot();
    expect(wrapper).to.exist();
  });

  it('render staton name on html', () => {
    expect(wrapper.children().length).to.equal(1);
    expect(wrapper.children().children().length).to.equal(1);
    expect(wrapper.children().children().text()).to.equal('Zywiec')
  });

  describe('onPress', () => {
    it('change closeDrawer', () => {
      expect(closeDrawerSpy).to.be.an('undefined');
      wrapper.find('TouchableHighlight').simulate('press');
      expect(closeDrawerSpy).to.equal(1);
    });
  });

  describe('#isActive', () => {
    it('return true', () => {
      expect(wrapper.instance().isActive()).to.equal(true)
    });

    it('return false', () => {
      wrapper.setProps({ getCurrentStationId: () => 2 });
      expect(wrapper.instance().isActive()).to.equal(false)
    });
  });
});
