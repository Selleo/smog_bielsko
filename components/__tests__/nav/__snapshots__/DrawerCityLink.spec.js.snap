exports[`DrawerCityLink renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    style={
        Object {
            "backgroundColor": "#aca",
          }
    }
    underlayColor="#aca">
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#444",
                "fontSize": 16,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
        }>
        Zywiec
    </Text>
</TouchableHighlight>,
  "nodes": Array [
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      style={
            Object {
                  "backgroundColor": "#aca",
                }
      }
      underlayColor="#aca">
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "color": "#444",
                        "fontSize": 16,
                        "paddingHorizontal": 20,
                        "paddingVertical": 10,
                      }
            }>
            Zywiec
      </Text>
</TouchableHighlight>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DrawerCityLink
        closeDrawer={[Function]}
        getCurrentStationId={[Function]}
        station={
                Object {
                        "id": 1,
                        "name": "Zywiec",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DrawerCityLink {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "closeDrawer": [Function],
          "getCurrentStationId": [Function],
          "station": Object {
            "id": 1,
            "name": "Zywiec",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          style={
                    Object {
                              "backgroundColor": "#aca",
                            }
          }
          underlayColor="#aca">
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#444",
                                        "fontSize": 16,
                                        "paddingHorizontal": 20,
                                        "paddingVertical": 10,
                                      }
                    }>
                    Zywiec
          </Text>
</TouchableHighlight>,
        "_debugID": 2,
        "_renderedOutput": <TouchableHighlight
          activeOpacity={0.85}
          onPress={[Function]}
          style={
                    Object {
                              "backgroundColor": "#aca",
                            }
          }
          underlayColor="#aca">
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#444",
                                        "fontSize": 16,
                                        "paddingHorizontal": 20,
                                        "paddingVertical": 10,
                                      }
                    }>
                    Zywiec
          </Text>
</TouchableHighlight>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DrawerCityLink
    closeDrawer={[Function]}
    getCurrentStationId={[Function]}
    station={
        Object {
            "id": 1,
            "name": "Zywiec",
          }
    } />,
}
`;
